src/scipp/compat/dict.py

```diff
@@ -184,18 +187,18 @@ def _dict_to_variable(d):
     return var
 
 
-def _dict_to_data_array(d):
+def _dict_to_data_array(d: dict[str, Any]) -> DataArray:
     """Convert a Python dict to a Scipp DataArray."""
     d = dict(d)
     if "data" not in d:
         raise KeyError(
             "To create a DataArray, the supplied dict must contain "
             f"'data'. Got {d.keys()}."
         )
-    out = {"coords": {}, "masks": {}, "attrs": {}}
-    for key in out.keys():
+    meta: dict[str, dict[str, Variable]] = {"coords": {}, "masks": {}, "attrs": {}}
+    for key in meta.keys():
```

Comment:
Interesting we are still using the deprecated attribute `meta` here?

Reply:
We aren't? `meta` is a local variable. But we do still use `attrs` which we should now remove.